libsolidity/interface/CompilerStack.cpp

```diff
@@ -1143,6 +1166,11 @@ SourceUnit const& CompilerStack::ast(std::string const& _sourceName) const
 	if (!source(_sourceName).ast)
 		solThrow(CompilerError, "Parsing was not successful.");
 
+	solAssert(
+		!isAnySourceExperimentalSolidity(),
+		"Experimental solidity does not support ast json output yet."
```

Comment:
Is AST actually unsupported? We definitely can produce an AST because the experimental analysis works fine in tests. We just don't run most of the original analysis steps.

I mean, it's possible that some things that use the AST fail because they try to access annotations that have not been filled, but I'd say it's those things that should have the assert and not the AST itself.

Reply:
I am still checking what crashes and what annotations are critical. At first, I tried to localize the assert on the failing/non-existent annotations, but that was taking too long and then I decided to just assert at the "root", so to speak...
I will try to make it more fine-grained.

Reply:
What happened to the idea of making a dependency graph for those annotations and outputs? You could take `ASTAnnotations.h`, take all the nullable annotations and grep the code to see which analysis stages and components read them. I see something around 25 such annotations, so it's a bit tedious but should be doable.

Reply:
What? An AST is of course fully unsupported :-).

Reply:
We cannot produce an AST for experimental solidity at all - there is zero code for handling it on export or import and that's not an oversight, but by design - the AST will likely change a lot before it is finalized, so dealing with it now is a waste of effort.
So definitely just throw an error if an AST is requested.

Reply:
To be clear: also any partial AST that we *may* be able to produce, would be fully misleading until we freeze things and handle them properly.

Reply:
What I meant was that we have something to return from `ast()` even on experimental. It may be missing some annotations, but it's still an AST that may be internally usable for some purposes. It's other things that depend on it that might depend on those missing annotations, like AST export to JSON, and I'd put the asserts there instead.